core/shared/src/main/scala/zio/stm/ZSTM.scala

```diff
@@ -381,6 +387,14 @@ final class ZSTM[-R, +E, +A] private[stm] (
       case TExit.Retry      => ZSTM.retry
     }
 
+  /**
+   * Propagates the success value to the first element of a tuple, but
+   * passes the effect input `R` along unmodified as the second element
+   * of the tuple.
+   */
+  def onFirst[R1 <: R, A1 >: A]: ZSTM[R1, E, (A1, R1)] =
```

Comment:
You don't need the `A` parameter here. I realize it was this way in `ZIO.scala` as well. We can either correct it as part of this PR or open a separate issue.